proselint/tools.py

```diff
@@ -365,6 +365,52 @@ def existence_check(text, list, err, msg, ignore_case=True, str=False,
     return errors
 
 
+def _case_sensitive_allowed_word(permitted: set[str], match: re.Match):
+    """Determine if a match object result is in a set of strings."""
+    matched = match.string[match.start():match.end()]
+    return matched in permitted
+
+
+def _case_insensitive_allowed_word(permitted: set[str], match: re.Match):
+    """Determine if a match object result is in a set, ignoring case."""
+    matched = match.string[match.start():match.end()].lower()
+    return matched in permitted
+
+
+def reverse_existence_check(
+        text, list, err, msg, ignore_case=True, offset=0
+        ):
+    """Find all words in ``text`` that aren't on the ``list``."""
+    if ignore_case:
+        permitted = set([word.lower() for word in list])
+        allowed_word = functools.partial(
+            _case_insensitive_allowed_word, permitted)
+    else:
+        permitted = set(list)
+        allowed_word = functools.partial(
+            _case_sensitive_allowed_word, permitted
+        )
```

Comment:
if my previous comment is followed, this can also be simplified.
```suggestion
    permitted = set([word.lower() for word in list] if ignore_case else list)
    allowed_word = functools.partial(
        _allowed_word, permitted, ignore_case=ignore_case
    )
```

Reply:
Oh, actually I missed your set-conversion tweak... hm, that feels a bit "fun" to parse (for humans), compared to:
```python
    if ignore_case:
        permitted = set([word.lower() for word in list])
    else:
        permitted = set(list)
```

I'm _open_ to it, just not sure it's necessary to make future code readers work that hard. Thoughts?

Reply:
i have never really considered the readability of inline if statements for Python programmers, i just know ternaries are a popular construct in other languages i use, like Rust and TypeScript. additionally, ruff has [a rule](https://docs.astral.sh/ruff/rules/if-else-block-instead-of-if-exp/) (derived from flake8-simplify) that prefers ternaries, so i would imagine they're not uncommon in Python as well.

Reply:
I tend to reserve them for situations where there's no other choice, like in list comprehensions. (Which can be downright squirrely to read under any circumstances; see my last commit that adds some indentation to try and make my whopper at least _semi_-readable.) I guess it all comes down to personal comfort; I know people who dislike code like this:
```python
if something:
    return one thing
return the other
```
...and always want the explicit `else:` in there. And then OTOH there are linters that will _flag_ the explicit `else:` as unnecessary (which of course it _is_, technically).

`¯\_(ツ)_/¯`